app/src/main/java/org/alphatilesapps/alphatiles/Validator.java

```diff
@@ -0,0 +1,992 @@
+
+                if (tileCounter >= 10) {
+                    if (tileCounter > 15) {
+                        fatalErrors.add("the word " + word + " in wordlist takes more than 15 tiles to build");
+                    } else {
+                        longWords += 1;
+                    }
+                }
+            }
+            if (longWords > 0) {
+                warnings.add("the wordlist has " + longWords + " long words (10 to 15 game tiles);" +
+                        " shorter words are preferable in an early literacy game. Consider removing longer words ");
+            }
+            for (Map.Entry<String, Integer> tile : tileUsage.entrySet()) {
+                if (tile.getValue() < 3) {
+                    warnings.add("the tile " + tile.getKey() + " in gametiles only appears in words " + tile.getValue()
+                            + " times. It is recommended that each letter be used in at least 3 words");
+                }
+            }
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+        try {
+            //todo.. the correct uppercase value?
+            Tab gameTiles = langPackGoogleSheet.getTabFromName("gametiles");
+            // below helper methods adds warnings for any duplicates in the provided column
+            gameTiles.checkColForDuplicates(0);
+            // the below portion makes sure each tile and their alternates are all unique
+            for (ArrayList<String> row : gameTiles) {
+                List<String> alternates = row.subList(0, 4);
+                if (new HashSet<>(alternates).size() < alternates.size()) {
+                    warnings.add("the row " + row + " in gametiles has the same tile appearing in multiple places");
+                }
+            }
+            // this check makes sure that colum 4 of gameTiles only has valid type specifiers
+            ArrayList<String> gameTileTypes = gameTiles.getCol(4);
+            HashSet<String> validTypes = new HashSet<>(Set.of("C","V","X", "AD", "AV", "BV", "FV", "LV"));
+            for (int i = 0; i < gameTileTypes.size(); i++) {
+                if (!validTypes.contains(gameTileTypes.get(i))) {
+                    warnings.add("row " + i + 1 + " of gametiles does not specify a valid type in the types column. Valid" +
+                            "types are " + validTypes);
+                }
+            }
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+        try {
+            langPackGoogleSheet.getTabFromName("wordlist").checkColForDuplicates(0);
+            langPackGoogleSheet.getTabFromName("wordlist").checkColForDuplicates(1);
+        } catch (ValidatorException e) {
+            warnings.add(GENERIC_WARNING);
+        }
+        try {
+            langPackGoogleSheet.getTabFromName("settings").checkColForDuplicates(0);
+            langPackGoogleSheet.getTabFromName("settings").checkColForDuplicates(1);
```

Comment:
Column 1 in the settings tab might have duplicates sometimes and that's ok (duplicate TRUEs, FALSEs, number values)
```suggestion
```